### DISCUSSION TITLE
Is it possible to get notification of a preference being toggled?

### DISCUSSION THREAD

**Discussion by pcolmer** [cloudscape-design/components#3082] [Q&A] [answered]
We have a table with a number of columns, some of which are grouped together. In the preferences, the user has the option of hiding an entire group, or just specific columns in a group:
![image](https://github.com/user-attachments/assets/507699e3-7fb0-4b6c-89a2-0e3c7de7205d)
However, if I turn off a group (e.g. Distribution), I'd _like_ to get an event or similar for that being toggled so that I can turn off all of the other columns associated with that group, e.g. Distribution - Fedora, Distribution - OpenEuler, DIstribution - Yocto.

Is that possible?

**Comment by cansuaa** [comment, marked as answer]
Hi Philip, currently only [onConfirm](https://cloudscape.design/components/collection-preferences/?tabId=api#events) event carries the values which is called when the Confirm button is clicked.

**Comment by pcolmer** [comment]
@cansuaa Thank you for that confirmation. Would it be reasonable to request some form of event for the scenario I've described, or does that go against the intention of that component?

**Comment by cansuaa** [comment]
Hi, I will bring this up with the team.

**Comment by jperals** [comment]
Hi Philip,

The behavior you describe would work best with visibility toggles at the group level, which is not a supported feature at the moment either. In any case, I took note of this use case in our internal tracking. Note however that I can't guarantee whether or when this might be implemented.

**Reply by pcolmer** [reply]
Thank you. I do understand that it might not be implemented ... but it would be much appreciated if it could be 😄
